src/rfxencode_diff_rlgr1.c

```diff
@@ -0,0 +1,281 @@
+/**
+ * FreeRDP: A Remote Desktop Protocol client.
+ * RemoteFX Codec Library - RLGR
+ *
+ * Copyright 2011 Vic Lee
+ * Copyright 2016-2017 Jay Sorg <jay.sorg@gmail.com>
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * This implementation of RLGR refers to
+ * [MS-RDPRFX] 3.1.8.1.7.3 RLGR1/RLGR3 Pseudocode
+ */
+
+#if defined(HAVE_CONFIG_H)
+#include <config_ac.h>
+#endif
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+
+#include "rfxcommon.h"
+
+#define GetNextInput do { \
+    input = *coef; \
+    coef++; \
+    coef_size--; \
+} while (0)
+
+#define CheckWrite do { \
+    while (bit_count >= 8) \
+    { \
+        bit_count -= 8; \
+        *cdata = bits >> bit_count; \
+        cdata++; \
+    } \
+} while (0)
+
+/*CodeGR(krp, lmag);*/ /* output GR code for (mag - 1) */
+#define CodeGR(_krp, _lmag) do { \
```

Comment:
In general, absolutely this should be a function.  In this case, speed is crucial and even the cost of a function call is considered.  Inline is an option but I've never liked it because it's a compiler suggestion.